client/src/pages/List.js

```diff
@@ -33,53 +44,126 @@ const useStyles = makeStyles({
+  }
 });
+const sampleArr1 = [{title:"art",status: "red", note: "March 14"}, {title:"philosophy",status: "red"}, {title:"cooking",status: "red"}, {title:"gym",status: "red"}]
+const sampleArr2 = [{title:"math", status: "green"}, {title:"english",status: "green"}, {title:"science",status: "green"}, {title:"history",status: "green"}]
+const columnTypes = {
+  ToDo: {
+    name: "To do",
+    items: sampleArr1,
+  },
+  InProgress: {
+    name: "In progress",
+    items: sampleArr2,
+  },
+};
 
 function List() {
   const classes = useStyles();
-  const examples = ["math", "english", "science", "history"];
-  const [list, setList] = useState(examples);
+  const [columns, setColumns] = useState(columnTypes);
+
+  function handleOnDragEnd(result, columns, setColumns) {
+    if (!result.destination) return;
+    const { source, destination } = result;
 
-  function handleOnDragEnd(res) {
-    if (!res.destination) return;
-    const items = Array.from(list);
-    const [reorderedItem] = items.splice(res.source.index, 1);
-    items.splice(res.destination.index, 0, reorderedItem);
-    setList(items);
+    if (source.droppableId !== destination.droppableId) {
+      const sourceColumn = columns[source.droppableId];
+      const destinationColumn = columns[destination.droppableId];
+      const sourceItems = [...sourceColumn.items];
+      const destinationItems = [...destinationColumn.items];
+      const [removed] = sourceItems.splice(source.index, 1);
+      destinationItems.splice(destination.index, 0, removed);
+      setColumns({
+        ...columns,
+        [source.droppableId]: {
+          ...sourceColumn,
+          items: sourceItems,
+        },
+        [destination.droppableId]: {
+          ...destinationColumn,
+          items: destinationItems,
+        },
+      });
+    } else {
+      const column = columns[source.droppableId];
+      const copiedItems = [...column.items];
+      const [removed] = copiedItems.splice(source.index, 1);
+      copiedItems.splice(destination.index, 0, removed);
+      setColumns({
+        ...columns,
+        [source.droppableId]: {
+          ...column,
+          items: copiedItems,
+        },
+      });
+    }
   }
 
   return (
     <div className={classes.mainContainer}>
       <Typography variant="h1">List</Typography>
-      <DragDropContext onDragEnd={handleOnDragEnd}>
-        <Droppable droppableId="column">
-          {(provided) => (
-            <Grid
-              className={classes.column}
-              container
-              direction="column"
-              justify="center"
-              {...provided.droppableProps}
-              ref={provided.innerRef}>
-              {list.map((e, idx) => (
-                <Draggable key={e} draggableId={e} index={idx}>
+      <DragDropContext
+        onDragEnd={(result) => handleOnDragEnd(result, columns, setColumns)}>
+        <div className={classes.columnsContainer}>
+          {Object.entries(columns).map(([id, column]) => {
```

Comment:
why don't we keep columns as an array of objects if we have to do Object.entries anyways? This can also help us to  keep track of their order

Reply:
An object format would allow me to use ".droppableId" as a way to correctly select the source & destination columns. I'm not sure how I would achieve that if it were an array format